tests/test_warm_reboot.py

```diff
@@ -234,7 +234,121 @@ def ping_new_ips(dvs):
+                                      ("ttl", ttl),
+                                      ("queue", queue)])
+    tbl.set(name, fvs)
+    time.sleep(1)
+
+
+def remove_mirror_session(dvs, name):
+    """
+        Deletes a mirror session. This method will wait 1 second to allow
+        redis updates to propagate.
+    """
+    tbl = swsscommon.Table(dvs.cdb, "MIRROR_SESSION")
+    tbl._del(name)
+    time.sleep(1)
+
+
+def get_mirror_session_state(dvs, name):
+    """
+        Gets the current state of a mirror session. This method will cause
+        your test to fail if the given mirror session does not exist.
+
+        Example output:
+        {
+            "src_ip": "10.0.0.0",
+            "dst_ip": "192.168.1.1",
+            ...
+        }
+    """
+    tbl = swsscommon.Table(dvs.sdb, "MIRROR_SESSION_TABLE")
+    (status, fvs) = tbl.get(name)
+    assert status
+    assert len(fvs) > 0
+    return {fv[0]: fv[1] for fv in fvs}
+
+
 class TestWarmReboot(object):
+    def test_MirrorSessionWarmReboot(self, dvs, testlog):
+        """
+            This test verifies that mirror session monitor ports are not
+            changed after warm reboot.
+        """
+
+        # Setup the virtual switch
+        dvs.setup_db()
+        dvs.runcmd("config warm_restart enable swss")
+
+        # Setup neighbors for the test
+        dvs.set_interface_status("Ethernet12", "up")
+        dvs.set_interface_status("Ethernet16", "up")
+        dvs.set_interface_status("Ethernet20", "up")
+
+        dvs.add_ip_address("Ethernet12", "10.0.0.0/31")
+        dvs.add_ip_address("Ethernet16", "11.0.0.0/31")
+        dvs.add_ip_address("Ethernet20", "12.0.0.0/31")
+
+        dvs.add_neighbor("Ethernet12", "10.0.0.1", "02:04:06:08:10:12")
+        dvs.add_neighbor("Ethernet16", "11.0.0.1", "03:04:06:08:10:12")
+        dvs.add_neighbor("Ethernet20", "12.0.0.1", "04:04:06:08:10:12")
+
+        # Set test_session1 monitor port as Ethernet12
+        dvs.add_route("2.2.2.2", "10.0.0.1")
+        create_mirror_session(dvs, "test_session1", "1.1.1.1", "2.2.2.2", "0x6558", "8", "100", "0")
+
+        # Set test_session2 monitor port as Ethernet16
+        dvs.update_route_ecmp("2.2.2.2", ["11.0.0.1", "10.0.0.1"])
+        create_mirror_session(dvs, "test_session2", "1.1.1.1", "2.2.2.2", "0x6558", "8", "100", "0")
+
+        # Set test_session3 monitor port as Ethernet20
+        dvs.update_route_ecmp("2.2.2.2", ["12.0.0.1", "11.0.0.1", "10.0.0.1"])
+        create_mirror_session(dvs, "test_session3", "1.1.1.1", "2.2.2.2", "0x6558", "8", "100", "0")
+
+        test_session1_mp = get_mirror_session_state(dvs, "test_session1")["monitor_port"]
+        test_session2_mp = get_mirror_session_state(dvs, "test_session2")["monitor_port"]
+        test_session3_mp = get_mirror_session_state(dvs, "test_session3")["monitor_port"]
+
+        # Perform the warm reboot
+        dvs.stop_swss()
+        dvs.start_swss()
+        time.sleep(5)
```

Comment:
is sleep 5 seconds enough? You could polling "show warm_restart state" and wait until orchagent is reconciled. That would take lot longer than 5 seconds though.